pkg/manager/manager_test.go

```diff
@@ -1858,6 +1852,54 @@ var _ = Describe("manger.Manager", func() {
 		Eventually(func() error { return goleak.Find(currentGRs) }).Should(Succeed())
 	})
 
+	It("should not leak goroutines if the default event broadcaster is used & events are emitted", func(specCtx SpecContext) {
```

Comment:
@sbueringer I have been debugging this test and when this suite is executed in isolation (`go test -v -race ./pkg/manager --ginkgo.fail-fast`) is succeeds, and so does the other test added to `internal/recorder/recorder_integration_test.go`.

It seems to me I am hitting some global timeout (see last Prow run) and am not sure how to best proceed here.

Reply:
> panic: test timed out after 10m0s
> FAIL	sigs.k8s.io/controller-runtime/pkg/internal/controller	600.306s
> https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/kubernetes-sigs_controller-runtime/3262/pull-controller-runtime-test/1956103172929884160

Looks lke the pkg/internal/controller test is timing out after 10m

Reply:
I took a quick look. I'm wondering if the tests are stuck here: https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/internal/controller/controller_test.go#L1501

Reply:
That might also be just a flake independent of your PR to be honest

Reply:
This manager leak detection test is now fixed; the event "search" logic was for the old API, making the test get stuck.

Now only the recorder integration test needs to be fixed and the PR should get green.